cpg_infra/driver.py

```diff
@@ -960,7 +960,40 @@ def setup_container_registry(self):
         + cpg-common container registries
         :return:
         """
+        self.setup_dataset_container_registry()
+        self.setup_inherited_container_registries()
 
+    def setup_dataset_container_registry(self):
+        """
+        If required, setup a container registry for a dataset
+        :return:
+        """
+        if not self.dataset_config.create_container_registry:
+            return
+
+        # mostly because this current format requires the project_id
+        custom_container_registry = self.infra.create_container_registry('images')
+        for kind, account in self.access_level_groups:
+            self.infra.add_member_to_container_registry(
+                f'test-images-reader-in-container-registry',
+                registry=custom_container_registry,
+                member=account,
+                membership=ContainerRegistryMembership.READER,
+            )
+            if kind in ('standard', 'full'):
+                self.infra.add_member_to_container_registry(
+                    f'{kind}-images-writer-in-container-registry',
+                    registry=custom_container_registry,
+                    member=account,
+                    membership=ContainerRegistryMembership.APPEND,
```

Comment:
Not directly related to this PR, but I just noted that `ContainerRegistryMembership.APPEND` is implemented as `role = 'roles/artifactregistry.writer'`, which seems to imply more than just appending? (I.e. also overwrite + delete?)
https://github.com/populationgenomics/cpg-infrastructure/blob/97d72179727f27a991a318ffae3e7b30990fe92c/cpg_infra/abstraction/gcp.py#L487-L488

This seems different from what we typically allow for `standard` (although we might want to revise that anyhow).

Reply:
Agreed, renamed to `writer`, although I believe you can't overwrite (as no delete) with writer: https://cloud.google.com/artifact-registry/docs/access-control